app/_components/TestComponent/BadComponent.tsx

```
@@ -0,0 +1,60 @@
+                <div key={item.id} onClick={() => inlineFunc(item.id)}>
+                    {item.name}
+                </div>
+            ))}
+
+            {/* Very bad: dangerouslySetInnerHTML */}
+            <div dangerouslySetInnerHTML={{ __html: htmlString }}></div>
+
+            {/* Inefficient way of referencing elements */}
+            <div ref={ref}></div>
+
+            <OldComponent />
```

ESLint

[error] 'OldComponent' is not defined.

(react/jsx-no-undef)